### PR TITLE
Add remaining siphon notebooks

### PR DESCRIPTION
**Pull request by lesserwhirls**
Add notebooks for Siphon (catalog basics and catalog filtering). Closes Unidata/pyaos-ams-2021#14, Unidata/pyaos-ams-2021#15, and Unidata/pyaos-ams-2021#16

### CODE REVIEW

File: notebooks/dataAccess/siphon-catalog-basics.ipynb [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+    "\n",
+    "---\n",
+    "\n",
+    "<div style=\"float:right; width:250 px\"><img src=\"../../instructors/images/siphon_tds_intro_preview.png\" alt=\"HTML view of a TDS Catalog\" style=\"height: 300px;\"></div>\n",
+    "\n",
+    "### Focuses\n",
+    "* Become familiar with THREDDS Catalogs and the THREDDS Data Server (TDS)\n",
+    "* Browse THREDDS Catalogs using siphon\n",
+    "* Show metadata and available datasets contained within a THREDDS Catalog\n",
+    "* List the data access methods assocaited with a dataset\n",
```
Comment (haileyajohnson) [inline notebooks/dataAccess/siphon-catalog-basics.ipynb:28]: **associated

File: notebooks/dataAccess/siphon-catalog-basics.ipynb [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+    "<div style=\"clear:both\"></div>\n",
+    "</div>\n",
+    "\n",
+    "---\n",
+    "\n",
+    "<div style=\"float:right; width:250 px\"><img src=\"../../instructors/images/siphon_tds_intro_preview.png\" alt=\"HTML view of a TDS Catalog\" style=\"height: 300px;\"></div>\n",
+    "\n",
+    "### Focuses\n",
+    "* Become familiar with THREDDS Catalogs and the THREDDS Data Server (TDS)\n",
+    "* Browse THREDDS Catalogs using siphon\n",
```
Comment (haileyajohnson) [inline notebooks/dataAccess/siphon-catalog-basics.ipynb:26]: Capitalize Siphon

File: notebooks/dataAccess/siphon-catalog-basics.ipynb [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+   "outputs": [],
+   "source": [
+    "catalog.catalog_refs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There are more things you can do once you have read a THREDDS Catalog using siphon, but for now we'll leave it at this.\n",
```
Comment (haileyajohnson) [inline notebooks/dataAccess/siphon-catalog-basics.ipynb:123]: Capitalize Siphon

File: notebooks/dataAccess/siphon-catalog-basics.ipynb [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+    "\n",
+    "<div class=\"alert alert-block alert-info\">\n",
+    "<b>Note: Not all TDS catalogs are intended to be browsed directly:</b> Occasionally, the TDS is used purely as \"middleware\", and the catalogs are not setup for users to easily browse directly.</div>\n",
+    "\n",
+    "An example of this would be the catalogs produced by the TDS serving data for the <b>N</b>orth <b>A</b>merica component of the <b>Co</b>ordinated <b>R</b>egional <b>D</b>ownscaling <b>Ex</b>periment ([NA-CORDEX](https://na-cordex.org/index.html)).\n",
+    "The intent of the data providers is for users to serch for datasets using the <a href=\"https://na-cordex.org/index.html\">NA-CORDEX search page</a> on the <a href=\"https://www.earthsystemgrid.org/\">NCAR Climate Data Gateway</a>, which allows for one to search for datasets by variable type, experiment, driver, model, etc.\n",
```
Comment (haileyajohnson) [inline notebooks/dataAccess/siphon-catalog-basics.ipynb:129]: **search

File: notebooks/dataAccess/siphon-catalog-basics.ipynb [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+    "Although the NA-CORDEX datasets are hosted on a TDS, they areall contained in one catalog, and their names are defined using a combination of the parameters used in the NA-CORDEX search page.\n",
```
Comment (haileyajohnson) [inline notebooks/dataAccess/siphon-catalog-basics.ipynb:130]: **are all

File: notebooks/dataAccess/siphon-catalog-basics.ipynb [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There are more things you can do once you have read a THREDDS Catalog using siphon, but for now we'll leave it at this.\n",
+    "\n",
+    "<div class=\"alert alert-block alert-info\">\n",
+    "<b>Note: Not all TDS catalogs are intended to be browsed directly:</b> Occasionally, the TDS is used purely as \"middleware\", and the catalogs are not setup for users to easily browse directly.</div>\n",
```
Comment (haileyajohnson) [inline notebooks/dataAccess/siphon-catalog-basics.ipynb:126]: The NA-CORDEX example is not showing up in the "Note" box, which makes it look like part of the tutorial and not part of an aside. Can you group the note and the example together?

File: notebooks/dataAccess/siphon-catalog-basics.ipynb [hunk 7]
```diff
@@ -0,0 +1,384 @@
...
+   "source": [
+    "from siphon.catalog import TDSCatalog"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "## Read a THREDDS Catalog from a TDS <a name=\"read\" />\n",
```
Comment (haileyajohnson) [inline notebooks/dataAccess/siphon-catalog-basics.ipynb:64]: Should the objective titles be numbered?

File: notebooks/dataAccess/siphon-catalog-basics.ipynb [hunk 8]
```diff
@@ -0,0 +1,384 @@
...
+   "outputs": [],
+   "source": [
+    "satellite_catalog.datasets"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `[]` indicate there are no datasets contained within the catalog.\n",
```
Comment (haileyajohnson) [inline notebooks/dataAccess/siphon-catalog-basics.ipynb:194]: ** indicates

File: notebooks/dataAccess/siphon-catalog-basics.ipynb [hunk 9]
```diff
@@ -0,0 +1,384 @@
...
+    "print('  datasets: {}\\n'.format(date_catalog.datasets))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We used the `follow()` method several times before finally reaching a catalog with datasets.\n",
+    "Normally, it is easiest to browse the catalogs of a TDS using a web browser in order to find a dataset collection that you might be interested in using.\n",
+    "One you have found a dataset you are interested in, you can use the URL from your browser to begin working in python using siphon.\n",
```
Comment (haileyajohnson) [inline notebooks/dataAccess/siphon-catalog-basics.ipynb:236]: ** Once you have, and capitalize Siphon

File: notebooks/dataAccess/siphon-catalog-basics.ipynb [hunk 10]
```diff
@@ -0,0 +1,384 @@
...
+   "source": [
+    "dataset.access_urls"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Each service provides a unique way of accessing the metadata or actual data contained within the dataset.\n",
+    "Other siphon notebooks explore ways in which the services can be used, but at this point, you are ready to begin your data analysis journey!"
```
Comment (haileyajohnson) [inline notebooks/dataAccess/siphon-catalog-basics.ipynb:308]: **Siphon

File: notebooks/dataAccess/siphon-catalog-filtering.ipynb [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+    "\n",
+    "### Objectives\n",
+    "1. [Find the Dataset within a THREDDS Catalog nearest to a given time](#filter_nearest)\n",
+    "1. [Find the Datasets within a THREDDS Catalog inside a given time range](#filter_range)\n",
+    "\n",
+    "---\n",
+    "\n",
+    "### Imports\n",
+    "\n",
+    "Often times a dataset name will contain useful information, like the datasets associated date and time.\n",
```
Comment (haileyajohnson) [inline notebooks/dataAccess/siphon-catalog-filtering.ipynb:35]: **dataset's

File: notebooks/dataAccess/siphon-catalog-filtering.ipynb [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+    "print('  minute: {}'.format (current_date_time.minute))\n",
+    "print('  second: {}'.format (current_date_time.second))\n",
+    "print('  microseconds: {}'.format(current_date_time.microsecond))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since we would like the catalog assocaied with the current `<year><month><day>`, we can use the _datetime_ object to make the appropriate string for use to use:"
```
Comment (haileyajohnson) [inline notebooks/dataAccess/siphon-catalog-filtering.ipynb:104]: **associated
** for us to use

File: notebooks/dataAccess/siphon-catalog-filtering.ipynb [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `strftime` method takes a format string (in this case, `'%Y%m%d'`).\n",
+    "This tells the `datetime` library how we want the date/time string to look when converted into a string.\n",
+    "`%Y` means use a four digit year, `%m` means use a two digit month, and `%d` means use a two digit day.\n",
+    "A full description of all of the possible formats can be found in the [python `datetime` documentation](https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes).\n",
+    "\n",
+    "Now that we have the catalog containing the data for today, lets examine the name of the first few datasets to get an idea of how they are named:"
```
Comment (haileyajohnson) [inline notebooks/dataAccess/siphon-catalog-filtering.ipynb:127]: **let's

File: notebooks/dataAccess/siphon-catalog-filtering.ipynb [hunk 14]
```diff
@@ -0,0 +1,239 @@
...
+    "datasets_for_today[0:3]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The dataset names should look like `Level3_Composite_n0r_1km_20210109_2355.gini`.\n",
+    "As we can see, the date and time is encoded within the filename.\n",
+    "When the filename contains a date and time in the form of `<year><month><day>_<hour><minute>`, siphon can automatically filter for times using the `filter_time_nearest` or `filter_time_range` methods.\n",
```
Comment (haileyajohnson) [inline notebooks/dataAccess/siphon-catalog-filtering.ipynb:146]: Siphon

File: notebooks/dataAccess/siphon-catalog-filtering.ipynb [hunk 15]
```diff
@@ -0,0 +1,239 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a href=\"#top\">Top</a>\n",
+    "\n",
+    "---\n",
+    "\n",
+    "## Find the Datasets within a THREDDS Catalog inside a given time range <a name=\"filter_range\" />\n",
+    "\n",
+    "We can use the `filter_time_range` method to find the datasets assocaited with the past hour by supplying start and end times:"
```
Comment (haileyajohnson) [inline notebooks/dataAccess/siphon-catalog-filtering.ipynb:173]: ** associated

**Review by haileyajohnson — CHANGES_REQUESTED** [review]
Looks good except for some typos, but nothing big.

**Comment by lesserwhirls** [comment]
Dang - I think I've hosed my Word dictionary, because it missed a lot of those issues! Updated the PR, and those should be fixed now. Thank you @haileyajohnson!

**Review by haileyajohnson — APPROVED** [review]

**Pull request merged**